extensions/data-workspace/src/main.ts

```diff
@@ -14,28 +14,32 @@ export function activate(context: vscode.ExtensionContext): void {
 	const workspaceService = new WorkspaceService();
 	const workspaceTreeDataProvider = new WorkspaceTreeDataProvider(workspaceService);
 	context.subscriptions.push(vscode.window.registerTreeDataProvider('dataworkspace.views.main', workspaceTreeDataProvider));
-	context.subscriptions.push(vscode.commands.registerCommand('projects.addProject', async () => {
+	context.subscriptions.push(vscode.commands.registerCommand('projects.addProject', async (uri: vscode.Uri) => {
 		// To Sakshi - You can replace the implementation with your complete dialog implementation
 		// but all the code here should be reusable by you
 		if (vscode.workspace.workspaceFile) {
-			const filters: { [name: string]: string[] } = {};
-			const projectTypes = await workspaceService.getAllProjectTypes();
-			filters[AllProjectTypes] = projectTypes.map(type => type.projectFileExtension);
-			projectTypes.forEach(type => {
-				filters[type.displayName] = [type.projectFileExtension];
-			});
-			let fileUris = await vscode.window.showOpenDialog({
-				canSelectFiles: true,
-				canSelectFolders: false,
-				canSelectMany: false,
-				defaultUri: vscode.Uri.file(path.dirname(vscode.workspace.workspaceFile.path)),
-				openLabel: SelectProjectFileActionName,
-				filters: filters
-			});
-			if (!fileUris || fileUris.length === 0) {
-				return;
+			if (uri) {
+				await workspaceService.addProjectsToWorkspace([uri]);
```

Comment:
in open from DB flow (and other flows) we will have vscode.workspace.workspaceFile ? as in the condition will be true?

Reply:
I think it is ok for Kim to assume a workspace already exists, Sakshi's work on the add project dialog should implement the workspace creation logic. and we can update this scenario after that.